### ISSUE TITLE
source and destination fs.state are getting same when run inside nexe

### ISSUE THREAD

**Issue by hnviradiya** [nexe/nexe#963]
if we use "[fs-extra](https://www.npmjs.com/package/fs-extra)"'s [copySync method](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/copy-sync.md). It gives source and destincation stat same in fs-extra.

Following is the line where "areIdentical" is getting to true when run inside nexe. But gives different without nexe (which should be actual expected behavior).

https://github.com/jprichardson/node-fs-extra/blob/69eb8964626871000de3a95548cd36035b5c4b27/lib/util/stat.js#L120

This issue only occurs when built with nexe. if we run without nexe it works fine.

OS: windows 11
NodeJs Version: v16.14.0

**Comment by calebboyd** [comment]
Thanks for the detailed report and the link to the problem code. Are you copying the binary itself? I'd think otherwise it would have a different inode. 

However, It might make sense to create a fake device for a virtual file stats..

**Comment by hnviradiya** [comment]
Not copying binary but js file. 

Following part of your comment, I didn't get though.
"However, It might make sense to create a fake device for a virtual file stats object. Right now it assumes the binary's device id."

**Comment by calebboyd** [comment]
Sorry that was an implementation detail, [relating to how application's files stat objects are faked.](https://github.com/nexe/nexe/blob/51b478a18d0e9fbcb67f9f6de4e6b0aec19be296/src/fs/patch.ts#L82)

Are the source and destination part of the program at build time?

**Comment by hnviradiya** [comment]
Yes. source we are including using --source option. and one thing I stated incorrectly in previous comment. that is, source is actually folder not js file. folder is basically one entire node package from node_modules.

**Comment by hnviradiya** [comment]
@calebboyd I had also logged issue on fs-extra and they closed it stating issue needs to be fixed from nexe side only. apart from that one more point, issue is only occurring in windows. Mac and linux are working fine. Moreover I figured when all files are inside "bin" folder then only this issue is occurring. If we put code at root level it is working fine.
